src/screens/WebsocketResearchScreen/SampleChatScreen.tsx

```diff
@@ -48,54 +53,72 @@ export const styles = StyleSheet.create({
 });
 
 const SampleChatScreen = () => {
-  const flatlistRef = React.useRef<FlatList>();
   const {
     selectedChannel,
     chats,
     goBackFromChatScreen,
     goToChatInfoScreen,
     sendChat,
-    updateChatContinuity
+    updateChatContinuity,
+    flatListRef: scrollRef,
+    scrollContext
   } = useChatScreenHook(ANONYMOUS);
-
+  const {replyPreview, clearReplyPreview} = useMessageHook();
   const updatedChats = updateChatContinuity(chats);
+  const [loading, setLoading] = React.useState(false);
+  const {anonProfileId} = useProfileHook();
+
+  const ownerChat = selectedChannel?.rawJson?.channel?.members?.find(
+    (item: any) => item.user_id === anonProfileId
+  );
+  const memberChat = selectedChannel?.rawJson?.channel?.members?.find(
+    (item: any) => item.user_id !== anonProfileId
+  );
 
   const renderChatItem = React.useCallback(({item, index}) => {
     return <BaseChatItem type={ANONYMOUS} item={item} index={index} />;
   }, []);
 
-  const scrollToEnd = () => {
-    flatlistRef.current?.scrollToEnd();
-  };
+  React.useEffect(() => {
+    return () => {
+      clearReplyPreview();
+    };
+  }, []);
 
   return (
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes here introduce new hooks and state variables to manage chat messages and user profiles. The use of `useEffect` to clear the reply preview when the component unmounts is a good practice. However, there seems to be a missing dependency in the `useEffect` hook at lines 82-86. The `clearReplyPreview` function should be included in the dependency array to ensure the effect runs correctly if this function changes.

```diff
-  React.useEffect(() => {
-    return () => {
-      clearReplyPreview();
-    };
-  }, []);
+  React.useEffect(() => {
+    return () => {
+      clearReplyPreview();
+    };
+  }, [clearReplyPreview]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->